src/js/modules/rpc/server.ts

```diff
@@ -31,10 +31,52 @@ import errors, { DisableResponseCode } from "./errors";
 import { Logger } from "winston";
 import Logging from "../utilities/Logging";
 import requireNative from "./require-native";
+import { isNtpEqual } from "../utilities/domain";
+import { find } from "../utilities/Map";
+
+function groupByTopic(
+  processBatchReplyItems: ProcessBatchReplyItem[]
+): ProcessBatchReplyItem[] {
+  interface Map {
+    key: [ProcessBatchReplyItem["ntp"], ProcessBatchReplyItem["coprocessorId"]];
+    value: ProcessBatchReplyItem;
+  }
+
+  const values = processBatchReplyItems
+    .reduce((prev, result) => {
+      const prevProcessBatch = find(
+        prev,
+        ([ntp, coprocId]) =>
+          isNtpEqual(result.ntp, ntp) && coprocId === result.coprocessorId
+      );
+
+      if (prevProcessBatch === undefined) {
+        return prev.set([result.ntp, result.coprocessorId], result);
+      } else {
+        const [id, prevResult] = prevProcessBatch;
+        if (
+          prevResult.resultRecordBatch === undefined ||
+          result.resultRecordBatch === undefined
+        ) {
+          return prev.set(id, { ...prevResult, resultRecordBatch: undefined });
```

Comment:
How could we ever fall into this case? We must have added an item to the collection but not a value. Is it possible that the code could arrive here?

Reply:
it can happen if the coprocessor fails and the policy error was Deregister in that case, we send the result as `undefined`